test/src/missle_launcher_test.cpp

```diff
@@ -5,22 +5,68 @@
 #include "mocks/mock_missile_launcher_listener.h"
 
 using ::testing::_;
+using ::testing::Property;
+using ::testing::InSequence;
 
-class MissleLauncherTest : public ::testing::Test {
+class MissileLauncherTest : public ::testing::Test {
 protected:
+    MissileLauncher* launcher;
+    MockMissileLauncherListener* launcher_listener;
+
     virtual void SetUp() {
+        launcher = new MissileLauncher() ;
+        launcher_listener = new MockMissileLauncherListener();
+        launcher->add_listener(launcher_listener);
     }
 
     virtual void TearDown() {
+        delete launcher_listener;
```

Comment:
I will look into why it does not report a memory leak then. But I am skeptical about deleting it because there is no memory leak error on compilation.